app/src/main/java/br/com/catbag/gifreduxsample/ui/GifListActivity.java

```diff
@@ -39,11 +38,25 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    protected void onStart() {
+        super.onStart();
+        mActionCreator.loadGifs();
+    }
+
+    @Override
+    public boolean hasStateChanged(AppState newState, AppState oldState) {
+        return !newState.equals(oldState);
+    }
+
+    @Override
     public void onStateChanged(AppState appState) {
-        if (!appState.getGifs().isEmpty()) {
+        if (!appState.getGifs().isEmpty() && mGifProgressVisibility) {
             mGifProgressVisibility = false;
+            anvilRender();
```

Comment:
In the activity actual scenario i put render inside to minimize unnecessary render because other components listen to StateChange and render when needed too. But move render to outside any condition on StateChange is a option to make easy our future code maintenance where we will not need any adjustment on render local. What do you think?